discovery-provider/plugins/notifications/src/email/notifications/index.ts

```diff
@@ -320,126 +350,126 @@ export async function processEmailNotifications(
-    const users = await getUsersCanNotify(identityDb, frequency, startOffset)
+    const pageCount = 1000
+    await getUsersCanNotify(identityDb, frequency, startOffset, pageCount, async (users) => {
+        if (Object.keys(users).length == 0) {
+          logger.info(`processEmailNotifications | No users to process. Exiting...`)
+          return
+        }
 
-    if (Object.keys(users).length == 0) {
-      logger.info(`processEmailNotifications | No users to process. Exiting...`)
-      return
-    }
+        logger.info(
+          `processEmailNotifications | Beginning processing ${
+            Object.keys(users).length
+          } users at ${frequency} frequency`
+        )
 
-    logger.info(
-      `processEmailNotifications | Beginning processing ${
-        Object.keys(users).length
-      } users at ${frequency} frequency`
+        await processGroupOfEmails(dnDb, identityDb, users, frequency, startOffset, remoteConfig)
+    })
+    
+  } catch (e) {
+    logger.error(
+      'processEmailNotifications | Error processing email notifications'
     )
+    logger.error(e)
+  }
+}
 
-    const userNotificationSettings = await buildUserNotificationSettings(
-      identityDb,
-      Object.keys(users).map(Number)
-    )
+const processGroupOfEmails = async (dnDb: Knex, identityDb : Knex, users: EmailUsers, frequency: EmailFrequency, startOffset: Moment, remoteConfig: RemoteConfig) => {
+  const userNotificationSettings = await buildUserNotificationSettings(
+    identityDb,
+    Object.keys(users).map(Number)
+  )
 
-    const notifications = await getNotifications(
-      dnDb,
-      frequency,
-      startOffset,
-      Object.keys(users),
-      remoteConfig
-    )
-    const groupedNotifications = groupNotifications(notifications, users)
+  const notifications = await getNotifications(
+    dnDb,
+    frequency,
+    startOffset,
+    Object.keys(users),
+    remoteConfig
+  )
+  const groupedNotifications = groupNotifications(notifications, users)
 
-    const currentUtcTime = moment.utc()
-    const chuckSize = 20
-    const results = []
-    let numEmailsSent = 0
-    for (
-      let chunk = 0;
-      chunk * chuckSize < groupedNotifications.length;
-      chunk += 1
-    ) {
-      const start = chunk * chuckSize
-      const end = (chunk + 1) * chuckSize
-      const chunkResults = await Promise.all(
-        groupedNotifications
-          .slice(start, end)
-          .map(async (userNotifications: UserEmailNotification) => {
-            try {
-              if (
-                !userNotificationSettings.shouldSendEmailAtFrequency({
-                  receiverUserId: userNotifications.user.blockchainUserId,
-                  frequency
-                })
-              ) {
-                return {
-                  result: Results.SHOULD_SKIP,
-                  error: 'User turned off or is abusive'
-                }
+  const currentUtcTime = moment.utc()
+  const chuckSize = 20
```

Comment:
tiny nit i just realized i think this is supposed to be chunk size but i like chuck size too lol